Fix installation issues for 130 and 131 and some other random stuff

tautomer opened the pull request:
Fixes #74 
Fixes #76 

1. Fix install.sh for v1.130 and v1.131
2. Fix function `clone_the_repo` when the repo does not already exist in `install.sh`
3. Parallelize `libheif` compilation in `pre-install.sh`
4. Fix `post-install.sh` for v1.131

tautomer committed:
Multiple fixes

1. Fix install.sh for v1.130 and v1.131
2. Fix function `clone_the_repo` when the repo does not already exist in
   `install.sh`
3. Parallelize `libheif` compilation in `pre-install.sh`
4. Fix `post-install.sh` for v1.131

moloroplayer commented on post-install.sh:
```diff
@@ -7,11 +7,30 @@ set -xeuo pipefail # Make my life easier
+    . ./.env
+    set +a
+}
+
```
load_environment_variables

moloroplayer replied:
this Line needs to be added that the variables will be loaded otherwise collum 28 failes because it can't find $REPO_TAG

tautomer committed:
Fix post-install

1. Environment variables were not loaded
2. & symbols were not escaped in the sed command

moloroplayer reviewed (approved):
Tested now all is Working (Y)

tautomer committed:
Use updated machine-learning/start.sh instead

The old machine-learning/start.sh can still be updated and reused for
v1.131, so I decided to revert the changes in post-install.sh to keep
the changes minimal.

tautomer commented:
> Tested now all is Working (Y)

Thanks! I also realized that the old machine-learning/start.sh could be updated and reused, so I chose to revert the changes made in post-install.sh. As long as start.sh is updated, the old systemd file will still work.

mrgpd-pfevre commented:
Hi there I'm also concerned by installing new immich LXC version.
I'm a bit confused with poetry vs uv which one should be used for ML build env ?

mrgpd-pfevre commented:
I've tried to switch to uv build env, but It brokes my ML service. about some cuda libs not found !
With poerty it was working like a charm for months of update.

tautomer commented:
> Hi there I'm also concerned by installing new immich LXC version. I'm a bit confused with poetry vs uv which one should be used for ML build env ?

In terms of installing packages, there should not be fundamental difference between these two, with uv being slightly faster based on my tests with immich. The speed up with uv is negligible (about 3 seconds) compared to the time spent on building dependencies and the server itself, so I kept poetry. 

> I've tried to switch to uv build env, but It brokes my ML service. about some cuda libs not found ! With poerty it was working like a charm for months of update.

Sorry, I do not have a device to test CUDA. You just did `uv sync --extra cuda` and the service is broken? I have no idea why this would happen... Maybe you can do a comparison between the packages installed with poetry vs uv. I believe both `poetry install` and `uv sync` relies on the same `pyproject.toml`, which should lead to the same environment.

mrgpd-pfevre commented:
Thx for your answer and details about poetry vs uv
I just fixed my problem, it was in fact my lxc cgroup settings that were no more in sync with the major device number of my graphic cards after reboot :-/
I have to find a way to fix proxmox nvidia kernel driver to change his device major number.
Regards,

scyotech commented:
Thanks @tautomer !!!   Beta-ed your 3 commits in my environment and upgraded to v1.131.3 with no problems!  Very much appreciated!

Spartan-II-117 commented:
@loeeeee can this be merged? I also tested it and it worked correctly.

loeeeee commented:
Thanks for all the time and efforts! :)
